pkg/service/did/ion.go

```diff
@@ -263,9 +263,10 @@ func (h *ionHandler) GetDIDs(ctx context.Context) (*GetDIDsResponse, error) {
 	}
 	dids := make([]did.Document, 0, len(gotDIDs))
 	for _, gotDID := range gotDIDs {
-		if !gotDID.IsSoftDeleted() {
+		if (deleted && gotDID.IsSoftDeleted()) || (!deleted && !gotDID.IsSoftDeleted()) {
```

Comment:
need to add this to key, web

Reply:
we should just move this logic to the DB I know @andresuribe87 has mentioned this before